DigitalLearningSolutions.Data/Services/SectionContentService.cs

```diff
@@ -24,23 +25,31 @@ public SectionContentService(IDbConnection connection, ILogger<SectionContentSer
 
         public SectionContent? GetSectionContent(int customisationId, int candidateId, int sectionId)
         {
-            return connection.QueryFirstOrDefault<SectionContent>(
+            SectionContent? sectionContent = null;
+            return connection.Query<SectionContent, SectionTutorial, SectionContent>(
                 @"
                     SELECT
-                        Customisations.CustomisationName,
                         Applications.ApplicationName,
+                        Customisations.CustomisationName,
                         Sections.SectionName,
-                        COALESCE(SUM(aspProgress.TutTime), 0) AS TimeMins, 
+                        COALESCE (aspProgress.TutTime, 0) AS TutTime,
                         Sections.AverageSectionMins AS AverageSectionTime, 
                         dbo.CheckCustomisationSectionHasLearning(Customisations.CustomisationID, Sections.SectionID) AS HasLearning,
-                        (CASE
-                            WHEN Progress.CandidateID IS NULL
-                                OR dbo.CheckCustomisationSectionHasLearning(Progress.CustomisationID, Sections.SectionID) = 0
-                            THEN 0
-                            ELSE CAST(SUM(aspProgress.TutStat) * 100 AS FLOAT) / (COUNT(Tutorials.TutorialID) * 2)
-                        END) AS PercentComplete
-                    FROM
-                        Tutorials
+                        COALESCE (aspProgress.DiagAttempts, 0) AS DiagAttempts,
+                        COALESCE (aspProgress.DiagLast, 0) AS DiagLast,
+                        Tutorials.DiagAssessOutOf,
+                        Sections.DiagAssessPath, 
+                        Sections.PLAssessPath,
+                        COALESCE (Attempts.AttemptsPL,0) AS AttemptsPL,
+                        COALESCE (Attempts.PLPassed,0) AS PLPassed,
```

Comment:
This and Attempts.PLPassed should be called `..Passes` instead of `...Passed` because of how we've changed the calculation.

Also the spacing is inconsistent here again and in one of the lines below.

And I think we're coalescing this twice now, both here and when defining Assessed? Both shouldn't be necessary. I think this one is necessary and the one when defining Assessed is not. Because you're doing a left join so the Attempts fields could be null. But when summing the AssessAttempts Status entries none of them can be null as that's not nullable 🙂